src/robot-scripts/webhook-listener.coffee

```diff
@@ -4,31 +4,46 @@
 
 AOHelper = require '../ActiveOwnerHelper'
 Review = require '../Review'
+Push = require '../Push'
 
 module.exports = (robot) ->
 
   helper = new AOHelper robot
+  branchPattern = ///^refs/heads/release-[0-9.]{5,6}$///
 
   robot.router.post '/hubot/gh', (req, res) ->
     robot.logger.debug 'GitHub webhook request received.'
+    test = req.body.ref
     res.send 200
-    return if ! req.body.pull_request
-    robot.logger.debug "Processing pull_request event. " +
-      "action: #{req.body.action}"
-    review = new Review(req.body)
-    if req.body.action == 'labeled' &&
-    req.body.label.name == process.env.HUBOT_REVIEW_NEEDED_LABEL
+    
+    return if ! (req.body.pull_request or req.body.forced == true)
+    #console.log(process.stdout.write req.body) this is failing on testing for unknown reasons. 
+    robot.logger.debug "Processing event. " +
+      "action: #{req.body}"
+
+    review = new Review(req.body) if req.body.pull_request
+    push = new Push(req.body) if req.body.forced
+
+    if req.body.action == 'labeled' and req.body.label.name == process.env.HUBOT_REVIEW_NEEDED_LABEL and req.body.pull_request
       robot.logger.debug "Emitted review-needed event for " +
       "#{req.body.pull_request.html_url}"
       robot.emit 'review-needed', review
-    if reviewNoLongerNeeded req.body
+
+    if req.body.pull_request and reviewNoLongerNeeded req.body
       robot.logger.debug "Emitted review-closed event for " +
       "#{req.body.pull_request.html_url}"
       robot.emit 'review-complete', review
 
+    #push needs to be the last choice
+    if req.body.ref == 'refs/heads/master' or test.match branchPattern
```

Comment:
This is checking if it's either mater or if it matches a regex pattern that should catch any release branch of 5 to 6 numbers. I'll need to lower that to 3 for release-4.0